HACKING.md

```diff
@@ -0,0 +1,95 @@
+# Hacking
+
+## Quickstart
+
+Use [ghcup][ghcup] to install the `cabal` cli tool and the ghc version we're using:
+
+```sh
+$ ghcup install ghc 8.10
+$ ghcup set ghc 8.10
+$ ghcup install cabal 
+$ ghcup set cabal
+$ cabal update
+$ cabal build
+```
+
+### Building
+
+In the base directory, run `cabal build`
+
+### Running tests
+
+In the base directory, run `cabal test`
+
+## Tools
+
+| name | description |
+| ---- | ----------- |
+| [ghcup][ghcup] | Used to manage installed versions of ghc and cabal-install |
+| ghc | The haskell compiler (installed via ghcup) |
+| cabal-install | The package manager we use (installed via ghcup) |
+| [haskell-language-server][hls] | LSP server for haskell projects |
+| [hlint][hlint] | A linting + hints tool for haskell code. It provides really useful suggestions |
+| [ormolu][ormolu] | A haskell source code formatter |
+| [fourmolu][fourmolu] | A forked version of ormolu that we are evaluating |
+
+### Installing haskell-language-server
+
+In VSCode: Install the "Haskell Language Server" (`haskell.haskell`) plugin in VSCode. In the
+settings there's a `Language Server Haskell: Hie Variant` option, which you can
+set to `haskell-language-server`
+
+## Linting
+
+Install [hlint][hlint].  Run `hlint /path/to/file-or-directory`.  You can also configure the haskell language server to run hlint.
+See [haskell-anguage-server][hls] for configuration instructions.
+
+`hlint` errors are usually required changes in pull requests, but we do not currently run `hlint` in CI, as there are a few outstanding lint errors that have not yet been fixed.
+You do not need to enforce that `hlint` passes to submit a PR, but it does help greatly, for both the author and reviewer.
+
+## Formatting
+
+Currently, we do not have a standardized formatting solution.  We have been using `ormolu`, but are now evaluating `fourmolu`, as it provides some configuration options that we want to take advantage of.
+
+For now, you can run `ormolu --mode inplace /path/to/file1 /path/to/file2`.  Shell globs and file-finders are recommended for formatting lots of files.  We do not require formatting yet, and when we do, we will ensure that it is present in CI.
```

Comment:
Rather than having folks install ormolu locally, we should recommend that folks use formatting through hls -- the vscode format-region and format-file options automatically proxy through hls